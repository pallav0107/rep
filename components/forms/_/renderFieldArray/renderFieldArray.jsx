import React from 'react';
import { IconAdd, IconClose } from '@wdpui/react-gel';
import { Field, FieldArray, reduxForm } from 'redux-form';
import { Flex, Box } from 'grid-styled';
import { InputForm } from '../renderInputField';
import { UIBlock, LIBlock } from '../../../../style/style';

const renderHobbies = ({ fields, meta: { error } }) => (
  <Flex>
    <UIBlock>
      {fields.map((hobby, index) => (
        <LIBlock key={index}>
          <Field
            name={hobby}
            type="text"
            component={InputForm}
            label={`Hobby #${index + 1}`}
          />
          <IconClose onClick={() => fields.remove(index)} mt={8} ml={5} />
        </LIBlock>
      ))}
      <li>
        <IconAdd onClick={() => fields.push()} />
        <span>Add Another Alternate name</span>
      </li>
      {error && <li className="error">{error}</li>}
    </UIBlock>
  </Flex>
);

const renderMembers = ({ fields, meta: { error, submitFailed } }) => (
  <Box>
    <Field
      name="another1"
      type="text"
      component={InputForm}
      label="First Name"
    />

    <FieldArray name="addmorenames" component={renderHobbies} />
  </Box>
);

const FieldArraysForm = props => {
  return <FieldArray name="members" component={renderMembers} />;
};

export default reduxForm({
  form: 'fieldArrays',
})(FieldArraysForm);
