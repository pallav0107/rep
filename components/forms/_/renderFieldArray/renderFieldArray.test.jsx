import React from 'react';
import { shallow } from 'enzyme';
import { FieldArraysForm } from './renderFieldArray';
import { withTheme } from '../../../../../utils/jest/TestUtils';
import "../../../../setupTests"

describe('renderHobbies', () => {
  it('should render', () => {
    const component = withTheme(<renderHobbies meta={['error']} />);
    const wrapper = shallow(component);
    expect(wrapper).toMatchSnapshot();
  });
});
describe('renderMembers', () => {
  it('should render', () => {
    const component = withTheme(
      <renderMembers meta={['error', 'submitFailed']} />
    );
    const wrapper = shallow(component);
    expect(wrapper).toMatchSnapshot();
  });
});
describe('FieldArraysForm', () => {
  it('should render', () => {
    const component = withTheme(<FieldArraysForm />);
    const wrapper = shallow(component);
    expect(wrapper).toMatchSnapshot();
  });
});
