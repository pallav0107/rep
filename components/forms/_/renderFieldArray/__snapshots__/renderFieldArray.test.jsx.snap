// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldArraysForm should render 1`] = `undefined`;

exports[`renderHobbies should render 1`] = `
<renderHobbies
  meta={
    Array [
      "error",
    ]
  }
/>
`;

exports[`renderMembers should render 1`] = `
<renderMembers
  meta={
    Array [
      "error",
      "submitFailed",
    ]
  }
/>
`;
